sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/RsaKeyLoaderScheduler.java

```diff
@@ -0,0 +1,51 @@
+package org.sdase.commons.server.auth.key;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.Validate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RsaKeyLoaderScheduler {
+  private static final Logger LOGGER = LoggerFactory.getLogger(RsaKeyLoaderScheduler.class);
+  private static final int DEFAULT_INITIAL_DELAY = 5;
+  private static final int DEFAULT_PERIOD = 5;
+  private static final TimeUnit DEFAULT_TIMEUNIT = TimeUnit.MINUTES;
+
+  private ScheduledExecutorService reloadKeysExecutorService;
+  private RsaPublicKeyLoader keyLoader;
+
+  private RsaKeyLoaderScheduler(
+      RsaPublicKeyLoader keyLoader, ScheduledExecutorService reloadKeysExecutorService) {
+    this.keyLoader = keyLoader;
+    this.reloadKeysExecutorService = reloadKeysExecutorService;
+  }
+
+  public static RsaKeyLoaderScheduler create(
+      RsaPublicKeyLoader keyLoader, ScheduledExecutorService reloadKeysExecutorService) {
+    Validate.notNull(keyLoader, "keyLoader should not be null");
+    Validate.notNull(reloadKeysExecutorService, "executorService should not be null");
+    return new RsaKeyLoaderScheduler(keyLoader, reloadKeysExecutorService);
+  }
+
+  public RsaKeyLoaderScheduler start(int initialDelay, int period, TimeUnit timeUnit) {
```

Comment:
Must this method check for something like isStarted and silently ignore it or stop and start again?
Or simpler: Can we make it package private and document „only call once“ so that nobody from outside can use it wrong?

Reply:
Done! I thought about setting it package-private as well!